lib/vmpooler/api/v1.rb

```diff
@@ -45,35 +45,39 @@ def alias_deref(hash)
       newhash
     end
 
-    def checkout_vm(template, result)
-      vm = backend.spop('vmpooler__ready__' + template)
+    def fetch_single_vm(template)
+      backend.spop('vmpooler__ready__' + template)
```

Comment:
Does this cause messages to be emitted to the log file? If so, I can imagine the log getting very noisy during periods of high load when there could be large numbers of failures for a given vm collection request.